help_to_heat/utils.py

```diff
@@ -249,3 +250,14 @@ def get_current_and_next_month_names(month_names):
 # this is to tell the user the first month we have no EPCs for, and the next month we expect to perform a dump
 def get_current_scottish_epc_cutoff_and_next_dump_month_names(month_names):
     return month_names[9], month_names[1]
+
+
+def get_most_recent_epc_per_uprn(address_and_lmk_details):
+    most_recent_address_and_lmk_details = []
+    address_and_lmk_details.sort(key=lambda x: x["uprn"], reverse=True)
+
+    for _, group in itertools.groupby(address_and_lmk_details, lambda x: x["uprn"]):
+        item = max(list(group), key=lambda x: datetime.strptime(x["lodgement-date"], "%Y-%m-%d"))
```

Comment:
does this work fine for epcs with no uprn?

Reply:
I didn't realise this was an issue, I'll add a case where this ignores no UPRN cases, since as discussed it hasn't been seen yet in practice, but is a niche possibility.

Reply:
for what its worth I've added this caveat to our docs here: https://softwiretech.atlassian.net/wiki/spaces/Support/pages/20562509848/DESNZ+GBIS+Help+to+Heat+EPCs#EPC-UPRNs